Better dev environment (#598)
open-telemetry/opentelemetry-demo: docs/development.md

# Development

> Development for this demo requires tooling in several different languages.

Reviewer: ```suggestion
Development for this demo requires tooling in several programming languages.
```

Minimum required versions will be noted where possible, but it is recommended
to update to the latest version for all tooling. The OpenTelemetry demo team

> will attempt the services in this repository upto date with the latest version

Reviewer: Nit:

```suggestion
will attempt the services in this repository up to date with the latest version
```

for dependencies and tooling when possible.

## Generate Protobuf files

The `make generate-protobuf` command is provided to generate protobuf files for

> all services. This can be used to compile code locally (without docker) and

Reviewer: ```suggestion
all services. This can be used to compile code locally (without Docker) and
```

> receive hints from IDEs such as IntelliJ or VS code.

Reviewer: ```suggestion
receive hints from IDEs such as IntelliJ or VS Code.
```

## Development tooling requirements

### .NET

- .NET Core runtime 6+

### C++

- build-essential
- pkg-config
- protobuf-compiler
- libprotobuf-dev
- libcurl4-openssl-dev
- nlohmann-json3-dev

> - cmake

Reviewer: Consider alphabetical ordering

### Elixir

- Erlang/OTP 23+
- Elixir 1.13+
- Rebar3 3.20+

### Go

- Go 1.19+

> - protoc 3.21+

Reviewer 2: To that to work I had to install the following:
```
go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
```
